src/main.rs

```diff
@@ -1,31 +1,39 @@
 #![allow(non_snake_case)]
-
+use clap::Parser;
 use std::{env, io, os::unix::process::CommandExt, process::Command};
 use syslog::{unix, Facility::LOG_AUTH, Formatter3164};
 
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    /// The program to run as root
+    program: String,
+    /// The program arguments
+    #[clap(default_value = "")]
+    args: String
```

Comment:
And this means application takes at most one argument and if none is given - it will pass "". Also no non utf8 arguments.